docs(k8s): enable disable ssh (#1465)
scaleway/docs-content: containers/kubernetes/how-to/enable-disable-ssh.mdx

---
meta:
  title: How to enable or disable SSH
  description: This page explains how to enable or disable the SSH service in your nodes
content:
  h1: How to enable or disable SSH
  paragraph: This page explains how to enable or disable the SSH service in your nodes
tags: kubernetes kapsule-cluster
dates:

>   validation: 2023-02-23

Author: ```suggestion
  validation: 2023-05-03
```

>   posted: 2023-02-23

Author: ```suggestion
  posted: 2023-05-02
```

categories:
  - containers
---

By default, an SSH server is installed and configured on the nodes of your Kubernetes Kapsule cluster.
To improve the security of your Kapsule cluster, you can disable this SSH server on your nodes.

You can manage and edit the status of the SSH server of your nodes from the [Scaleway console](https://console.scaleway.com) or by using the [API](https://developers.scaleway.com/en/).
Here, we show you how to do it from the console.

<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [created](/containers/kubernetes/how-to/create-cluster) a Kubernetes Kapsule cluster
  - [This feature is available on the nodes of your cluster](#how-to-know-if-this-feature-is-available-on-the-nodes-of-your-cluster)
</Message>

## How to enable or disable SSH on the nodes of a pool

1. Click **Kubernetes** in the **Containers** section of the [Scaleway Console](https://console.scaleway.com) side menu.
2. Click on the cluster you want to edit.
3. Click on the **Pools** tab.
   <Lightbox src="scaleway-kapsule_pools.webp" alt="" />

> 4. Click <Icon name="more" /> next to the pool you wish to update and click on **Edit**.

Reviewer: ```suggestion
4. Click <Icon name="more"/> next to the pool you wish to update and click on **Edit**.
```

5. Add the `ssh-enabled=false` tag to disable SSH or `ssh-enabled=true` to enable it. Make sure there is only one tag with the `ssh-enabled=` prefix.

> 6. Click on **Update pool**.

Reviewer: ```suggestion
6. Click **Update pool**.
```

7. Wait approximately 5 minutes until the tag is propagated to the nodes of your pool and the status of the SSH server of each node is updated.

## How to know if this feature is available on the nodes of your cluster

1. Open a terminal program.

> 2. Enter the command below into the terminal. Make sure you replace `path_to_your_kubeconfig` with the path to your kubeconfig file.

Reviewer: ```suggestion
2. Run the following command into the terminal. Make sure you replace `path_to_your_kubeconfig` with the path to your kubeconfig file.
```

    ```bash
    KUBECONFIG=path_to_your_kubeconfig kubectl get nodes -o custom-columns=NAME:.metadata.name,"AGENT-VERSION":".metadata.annotations.k8s\.scaleway\.com/agent-version"
    ```

    <Message type="tip">
      To find the kubeconfig of your Kapsule cluster, log into the Scaleway console and click **Kubernetes** in the **Containers** section of the side menu.

>       A list of your Kubernetes clusters will display, click on the one you want to verify. In the Download kubeconfig section at the bottom of the page, click on **Download file** to download the kubeconfig.

Reviewer 2: ```suggestion
      A list of your Kubernetes clusters will display, click on the one you want to verify. In the Download kubeconfig section at the bottom of the page, click **Download file** to download the kubeconfig.
```

>       A list of your Kubernetes clusters will display, click on the one you want to verify. In the Download kubeconfig section at the bottom of the page, click on **Download file** to download the kubeconfig.

Reviewer: ```suggestion
      A list of your Kubernetes clusters will display, click on the one you want to verify. In the Download kubeconfig section at the bottom of the page, click **Download file** to download the kubeconfig.
```
